packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -136,25 +137,29 @@ export class ApolloServerBase<Request = RequestInit> {
       this.createSubscriptionServer(this.http, config);
     }
 
-    if (opts.engine || opts.engineInRequestPath) this.createEngine(opts);
+    if (options.engineProxy) this.createEngine(options.engineProxy);
 
     return new Promise((success, fail) => {
-      if (this.engine) {
-        this.engine.listen(
-          Object.assign({}, options.engineLauncherOptions, {
-            graphqlPaths: [this.graphqlPath],
-            port: options.port,
-            httpServer: this.http,
-          }),
+      if (this.engineProxy) {
+        this.engineProxy.listen(
+          Object.assign(
+            {},
+            typeof options.engineProxy !== 'boolean' ? options.engineProxy : {},
```

Comment:
This is incorrect. You've conflated the "launcher options" with the "engine configuration". They are two different things.  See https://www.apollographql.com/docs/engine/setup-node.html#api-engine.listen

Though actually the code before this commit was wrong too.  `options.launcherOptions` should have been passed as an option called `launcherOptions` to `engine.listen`, not merged with the top level listen options.

Basically the semantics here are:
- ApolloEngine constructor options (or ApolloEngineLauncher constructor options): Literally the JSON config file that gets passed to the proxy unchanged.
- engine.listen() launcherOptions sub-field (identical to ApolloEngineLauncher.start options): Options relevant to the concept of running an external process.  Stuff like output redirection, signal handling, startup timeout.
- engine.listen() options:  Stuff specific to ApolloEngine's magic of stitching together your HTTP server and the proxy. So telling it what path and port to use, etc.  AS 2.0 actually explicitly sets all of them other than innerHost so it's probably OK to not let them be customized.